clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -422,6 +422,31 @@ LogicalResult CastOp::verify() {
   llvm_unreachable("Unknown CastOp kind?");
 }
 
+//===----------------------------------------------------------------------===//
+// VecValueOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult VecValueOp::verify() {
+  // Verify that the number of arguments matches the number of elements in the
```

Comment:
Note: you can also properly constraint the operations in tablegen (like I recently introduced for ints https://github.com/llvm/clangir/commit/f9cb8a5a70f54a65a27c263407ff9195bc735e55), that sometimes can allow us to skip the C++ implementation for verify.

Reply:
Same comments about planning to tackle TableGen improvements later.